server/plugins/client/bundle.js

```diff
@@ -14,20 +28,27 @@ var parallel = require('async').parallel
  * and avoiding unneeded bundling with browserify saves a significant time.
  */
 function bundleClient (hoodieClientPath, bundleTargetPath, config, callback) {
-  parallel([
+  var plugins = [
+    path.resolve(config.paths && config.paths.applicationRoot || '', 'hoodie/client')
+  ].filter(checkModule)
+  var getPluginsModifiedTimes = plugins.map(function (pluginPath) {
+    return getModifiedTime.bind(null, requireResolve(pluginPath))
+  })
```

Comment:
> No, it's an array [...] of bound instances of the getModifiedTime function

Ooops you are totally right, no idea how I could miss that. The name is good then, sorry for the confusion :)

> I think hooking into the browserify API may indeed be the way to go

Yeah I was thinking about saving the output of a JSON tree somehow. But it sounds fairly complicated, I wouldn’t mind to leave things simple as they are right now and document its limitation, with a link to an issue that describes it in detail and what the approach would be to fix it, so that we can work on this separately.

I think your plan sounds great :) A few more thoughts for the bundled hoodie.js client: 

1. Sometimes people  don’t need the bundling at all, because they bundle the client manually into their frontend application assets, so it would be nice to disable it altogether and return a helpful error message
2. The bundling should only be done in development. For deployment, we still need to figure out a way that would let us bundle the client and then make it part of the deployed server. I don’t yet know how to do it, so just want to put this thought out there :)

I think it makes sense to get your work so far merged first as it unblocks other important work, and then look into a proper implementation of the client bundle cache?